core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala

```diff
@@ -528,15 +528,22 @@ private[deploy] class SparkSubmitArguments(args: Seq[String], env: Map[String, S
     if (opt.startsWith("-")) {
       error(s"Unrecognized option '$opt'.")
     }
+    /** This allows for you to add variables for basic auth in this format:
+     * ${NOMAD_META_USER}:${NOMAD_META_PASS}
+     * See Nomad docs on Variable Interpolation for more information
+     * https://www.nomadproject.io/docs/runtime/interpolation
+     * */
+    val cleanOpt = opt.replaceAll("\\$\\{[a-zA-Z_.]*\\}:\\$\\{[a-zA-Z_.]*\\}\\@", "")
```

Comment:
is the idea that (when a user specifies an artylab jar) we will interpolate our service account credentials into the artylab URL in `mot-nomad-spark-agents` when we build the job template? if so, we create the `SparkSubmitArguments` before we call `NomadClusterModeConf.apply` so in that scenario we wouldn't need to remove the variables here

Reply:
Makes sense. Removed the code from SparkSubmit to keep this a bit more maintainable.